src/tests/Equinor.ProCoSys.IPO.WebApi.IntegrationTests/Invitations/InvitationsControllerTests.cs

```diff
@@ -40,20 +40,309 @@ public async Task GetInvitationsByCommPkgNo_AsViewer_ShouldGetInvitations()
+                .Single(p => p.Organization == Organization.TechnicalIntegrity).Person;
+
+            // Act
+            if (participantPerson.Person.Id != null)
+            {
+                await InvitationsControllerTestsHelper.SignPunchOutAsync(
+                    UserType.Signer,
+                    TestFactory.PlantWithAccess,
+                    invitationToSignId,
+                    (int)participantPerson.Person.Id,
+                    participantPerson.Person.RowVersion);
+            }
+
+            // Assert
+            var signedInvitation = await InvitationsControllerTestsHelper.GetInvitationAsync(
+                UserType.Viewer,
+                TestFactory.PlantWithAccess,
+                invitationToSignId);
+
+            var signerParticipant = signedInvitation.Participants.Single(p => p.Person?.Person.Id == participantPerson.Person.Id);
+            Assert.IsNotNull(signerParticipant.SignedAtUtc);
+            Assert.AreEqual("SigurdUserName", signerParticipant.SignedBy);
+        }
+
+        [TestMethod]
+        public async Task CompletePunchOut_AsCompleter_ShouldCompletePunchOut()
+        {
+            // Arrange
+            var invitationToCompleteId = await InvitationsControllerTestsHelper.CreateInvitationAsync(
+                UserType.Planner,
+                TestFactory.PlantWithAccess,
+                "InvitationForCompletingTitle",
+                "InvitationForCompletingDescription",
+                InvitationLocation,
+                DisciplineType.DP,
+                _invitationStartTime,
+                _invitationEndTime,
+                _participantsForSigning,
+                _mcPkgScope,
+                null
+            );
+
+            var invitation = await InvitationsControllerTestsHelper.GetInvitationAsync(
+                UserType.Signer,
+                TestFactory.PlantWithAccess,
+                invitationToCompleteId);
+
+            var completerPerson = invitation.Participants
+                .Single(p => p.Organization == Organization.Contractor).Person;
+
+            if (completerPerson.Person.Id != null)
+            {
+                var completePunchOutDto = new CompletePunchOutDto()
+                {
+                    InvitationRowVersion = invitation.RowVersion,
+                    ParticipantRowVersion = completerPerson.Person.RowVersion,
+                    Participants = new List<ParticipantToChangeDto>
+                    {
+                        new ParticipantToChangeDto
+                        {
+                            Id = (int)completerPerson.Person.Id,
+                            Note = "Some note about the punch round or attendee",
+                            RowVersion = completerPerson.Person.RowVersion,
+                            Attended = true
+                        }
+                    }
+                };
+
+            // Act
+            await InvitationsControllerTestsHelper.CompletePunchOutAsync(
```

Comment:
why do we not need RowVersion as input here? The commandhandler creates a new RowVersion so I guess we should Assert for changed RowVersion here too?